blockbuster/abci/abci.go

```diff
@@ -33,10 +34,22 @@ func NewProposalHandler(logger log.Logger, mempool blockbuster.Mempool) *Proposa
 // the default lane will not have a boundary on the number of bytes that can be included in the proposal and
 // will include all valid transactions in the proposal (up to MaxTxBytes).
 func (h *ProposalHandler) PrepareProposalHandler() sdk.PrepareProposalHandler {
-	return func(ctx sdk.Context, req abci.RequestPrepareProposal) abci.ResponsePrepareProposal {
+	return func(ctx sdk.Context, req abci.RequestPrepareProposal) (resp abci.ResponsePrepareProposal) {
+		// In the case where there is a panic, we recover here and return an empty proposal.
+		defer func() {
+			if err := recover(); err != nil {
+				h.logger.Error("failed to prepare proposal", "err", err)
+				resp = abci.ResponsePrepareProposal{Txs: make([][]byte, 0)}
+			}
+		}()
+
 		proposal := h.prepareLanesHandler(ctx, blockbuster.NewProposal(req.MaxTxBytes))
 
-		return abci.ResponsePrepareProposal{Txs: proposal.Txs}
+		resp = abci.ResponsePrepareProposal{
+			Txs: proposal.Txs,
+		}
+
+		return
```

Comment:
nit: pls no naked returns. They make debugging hell.

```suggestion
		return resp
```